### PR TITLE
Upgrade external dependencies' version to spring-boot 2.6.6

### PR DESCRIPTION
**Pull request by Netyyyy**
1. spring-boot:2.6.6 released: https://github.com/spring-projects/spring-boot/releases/tag/v2.6.6
2. spring-boot-dependencies: https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/2.6.6/spring-boot-dependencies-2.6.6.pom

Change:
1. pom.xml(azure-identity) :
    due to error message `Found Banned Dependency: net.minidev:json-smart:jar:2.4.7` , 
    found maven confilct between `<version>[1.3.3,2.4.7]</version>` and `<version>2.4.8</version>`,
    [FYI](https://guntherpopp.blogspot.com/2011/03/understanding-maven-dependency.html) ,version range has higher priority,
    decide not to change version of  `json-smart` to 2.4.8.

Note:
  Due to the delay in the release of spring-cloud:2021.0.2, we will keep spring-cloud:2021.0.1

### CODE REVIEW

**Comment by Netyyyy** [comment]
Hi @alzimmermsft and @rickle-msft, I checked [this](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1504210&view=logs&j=a4f6fe1f-26fc-5600-f4b6-3314d094d834&t=d40940cd-7c32-5d50-3cc5-c571523e551e&l=85359) in local with the verison2.6.6 and not updated version, both met these two errors :
image
, 
image
, It doesn't seem to be caused by the spring-boot version update, please help to fix it asap, thank you!

**Comment by alzimmermsft** [comment]
@Netyyyy this PR should resolve the error in `azure-storage-blob-nio`.

cc: @rickle-msft

**Comment by Netyyyy** [comment]
Hi @g2vinay , I made a [change](https://github.com/Azure/azure-sdk-for-java/pull/28280#issue-1202863972) in azure-identity dependency for spring-boot version 2.6.6, could you make sure that's fine for others.

**Comment by Netyyyy** [comment]
@kushagraThapar , these [errors](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1516720&view=results) seem to be caused by `diagnostics` tests, could we merge this [PR](https://github.com/Azure/azure-sdk-for-java/pull/28280) ignoring cosmos-test errors, and if not, when will these errors be fixed?

**Comment by Netyyyy** [comment]
@vcolin7 , I'm not sure what caused [keyvault-test](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1516721&view=results) failed, but I saw recently keyvault-test all failed with similar reason, so could we ignore these errors when merging this version update pr into main, and if not, could you help to fix it asap?

**Comment by kushagraThapar** [comment]
> @kushagraThapar , these [errors](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1516720&view=results) seem to be caused by `diagnostics` tests, could we merge this [PR](https://github.com/Azure/azure-sdk-for-java/pull/28280) ignoring these cosmos-test errors, and if not, when will these errors be fixed?

@Netyyyy - I verified, and the diagnostics test fix has been merged in the latest master, so you should be good now. Please pull the latest master in your branch and you can re-run the cosmos pipelines to verify, thanks!

**Comment by vcolin7** [comment]
> @vcolin7 , I'm not sure what caused [keyvault-test](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1516721&view=results) failed, but I saw recently keyvault-test all failed with similar reason, so could we ignore these errors when merging this version update pr into main, and if not, could you help to fix it asap?

It looks like a transient issue, sometimes dependencies cannot be retrieved from Maven for whatever reason. Usually re-running the failed jobs does the trick. I've started a re-run for this PR, let's see if the problem persists.

File: sdk/identity/azure-identity/pom.xml [hunk 1]
```diff
@@ -41,6 +41,12 @@
     <dependency>
       <groupId>com.microsoft.azure</groupId>
       <artifactId>msal4j</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>net.minidev</groupId>
+          <artifactId>json-smart</artifactId>
```
Comment (g2vinay) [inline sdk/identity/azure-identity/pom.xml:47]: This exclusion can result in customers running into dependency compatibility issues with json-smart version:

1. Exclusion will prevent maven from determining which json-smart version is compatible with oauth2-oidc-sdk in msal4j and by default follow the top level json-smart version that can be incompatible with oauth2-oidc-sdk.

2. Retaining the json-smart dependency in msal4j will let maven determine the compatible version with it automatically.

We should not exclude this dependency.

File: sdk/identity/azure-identity/pom.xml [hunk 2]
```diff
@@ -90,7 +96,7 @@
     <dependency>
       <groupId>net.minidev</groupId>
       <artifactId>json-smart</artifactId>
-      <version>2.4.7</version> <!-- {x-version-update;net.minidev:json-smart;external_dependency} -->
+      <version>2.4.8</version> <!-- {x-version-update;net.minidev:json-smart;external_dependency} -->
```
Comment (g2vinay) [inline sdk/identity/azure-identity/pom.xml:99]: msal4j currently uses version 2.4.7
We should ask msal4j to upgrade to latest oauth2-oidc-sdk first to ensure it is compatible with json-smart:2.4.8 before upgrading the pinned version here.

Comment (Netyyyy) [reply sdk/identity/azure-identity/pom.xml]: ok, will remain 2.4.7 this pr meanwhile ask msal4j to upgrade oauth2-oidc-sdk

**Comment by Netyyyy** [comment]
> 

https://github.com/Azure/azure-sdk-for-java/pull/28343

**Comment by Netyyyy** [comment]
> > @kushagraThapar
> >             
> >             Kushagra Thapar
> >             FTE , these [errors](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1516720&view=results) seem to be caused by `diagnostics` tests, could we merge this [PR](https://github.com/Azure/azure-sdk-for-java/pull/28280) ignoring these cosmos-test errors, and if not, when will these errors be fixed?
> 
> @Netyyyy Muyao Feng (CSI Interfusion Inc) Vendor - I verified, and the diagnostics test fix has been merged in the latest master, so you should be good now. Please pull the latest master in your branch and you can re-run the cosmos pipelines to verify, thanks!

should in this [PR](https://github.com/Azure/azure-sdk-for-java/pull/28343)?

**Comment by kushagraThapar** [comment]
> > > @kushagraThapar
> > > ```
> > >         Kushagra Thapar
> > >         FTE , these [errors](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1516720&view=results) seem to be caused by `diagnostics` tests, could we merge this [PR](https://github.com/Azure/azure-sdk-for-java/pull/28280) ignoring these cosmos-test errors, and if not, when will these errors be fixed?
> > > ```
> > 
> > 
> > @Netyyyy Muyao Feng (CSI Interfusion Inc) Vendor - I verified, and the diagnostics test fix has been merged in the latest master, so you should be good now. Please pull the latest master in your branch and you can re-run the cosmos pipelines to verify, thanks!
> 
> should in this [PR](https://github.com/Azure/azure-sdk-for-java/pull/28343)?

@Netyyyy - not this PR which you mentioned above. 
The cosmos diagnostics tests have been fixed, and I see your PR is good to go from Cosmos SDK point of view. The only tests failing on your PR currently are just flaky tests, you can ignore them. Thanks!

**Review by saragluna — APPROVED** [review]

**Pull request merged**